adafruit_httpserver.py

```diff
@@ -70,12 +67,12 @@ def __init__(
             self.method = method
         else:
             # Parse request data from raw request
-            match = self._REQUEST_RE.match(raw_request.decode("utf8"))
-            if match:
-                self.path = match.group(2)
-                self.method = match.group(1)
-            else:
-                raise ValueError("Unparseable raw_request:", raw_request)
+            request_text = raw_request.decode("utf8")
+            first_line = request_text[: request_text.find("\n")]
```

Comment:
```suggestion
            first_line = request_text[: request_text.find("\r")]
```
Request lines have to end in `\r\n`, so you may as well scan for `\r` here.

Reply:
Do clients always send "\r\n"? I went with the "\n" version because it handles both variants. (It's a bit non-obvious that it does handle both well.)

Reply:
(We haven't measured the speed improvement - but subjectively it seems much better.)

Reply:
Thinking more about this, if we want to be strict should we do: `request_text.find("\r\n")`?

Reply:
https://stackoverflow.com/questions/5757290/http-header-line-break-style discusses what the RFC's say. The standard says `\r\n`. If you are writing a tolerant client, you could look for only `\n`, but it's out of spec. So you could leave it as `\n`, but then I would change `_http_version` to just `_`, because otherwise it's the HTTP version with a trailing `\r`, in nearly all cases.

Browsers always send `\r\n`, because they want to follow the standard.

Reply:
Surprisingly, the split strips the trailing `\r`.
```python
>>> "GET / HTTP1.1\r".split()
['GET', '/', 'HTTP1.1']
```
That's what I meant by "a bit non-obvious". I should have clarified.

So, the choice is between a tolerant (but non-obvious) implementation, or a clear and strict implementation. If real world clients tend to follow the standard, then I agree we should split on `\r\n`.

Reply:
Argh! Sorry, I should know the semantics of Python `split()` better. I would then suggest you drop the new commit and go back to `\n`, since it's more tolerant.

Reply:
I've dropped the new commit.